caillef/dustzh.lua

```diff
@@ -257,7 +263,49 @@ Client.Action3Release = function()
 end
 
 
+uiControls = {}
+local uiControlsMetatable = {
+    __index = {
```

Comment:
There's no need add complexity with a metatable.
I see the intention was to mimic other pre-existing components in that script, but they should also be simplified. 😅

Also, I would prefer to use a naming like `instructions` instead of `uiControls`, it'd be more accurate. 
Here's how it could be: 

```lua
instructions = {
    bg = nil
}

instruction.display = function(self)
    if self.bg == nil then
        -- init self.bg here
    end
    self.bg:show()
end

instructions.hide = function(self)
    if self.bg ~= nil then self.bg:hide() end
end
```

Reply:
solved